scripts/helpers/normalize_helpers.rb

```diff
@@ -40,15 +42,21 @@ def sort_classes_strip_whitespace(element)
 # The index of copied elements (the number in _copy_23) isn't meaningful so
 # hide it.
 
-def mask_copied_id_numbers(element)
+def mask_copied_id_numbers(element, existing_ids)
   return unless element[:id]
 
+  ids_to_exclude = %w[author-1 publisher-1 publisher-2 copyright-holder-1]
+  if existing_ids.include?(element[:id]) && !ids_to_exclude.include?(element[:id])
+    puts "warning! duplicate id found for #{element[:id]}"
+  end
+  existing_ids.add(element[:id])
```

Comment:
Let's store the "ids to exclude" as a constant at the top of the file since it never changes

```ruby
DUPLICATE_IDS_TO_IGNORE = %w[author-1 publisher-1 publisher-2 copyright-holder-1]
```

Let's put this code into its own self-contained method -- it'll help us make this code clearer and it will be self-documenting

```ruby
def mask_copied_id_numbers(element)
  return unless element[:id]

  warn_if_id_already_seen(element[:id])
  element[:id] = element[:id].gsub(/_copy_(\d+)$/, '_copy_XXX')
end

def warn_if_id_already_seen(id)
  @already_seen_ids ||= Set.new
  if @already_seen_ids.include?(id) && !DUPLICATE_IDS_TO_IGNORE.include?(id)
    puts "warning! duplicate id found for #{id}"
  end
  @already_seen_ids.add(id)
end
```

Then in `normalize` below you can remove `ids_set = Set.new` and revert the change on line 59

Reply:
Thanks, will do! Regarding `DUPLICATE_IDS_TO_IGNORE`, I'm not sure _why_ those ids seem to be duplicated in the easybaked version -- could this be a bug in easybake that kitchen should fix?

Reply:
Yes could be.  Let's keep a list somewhere (maybe in Kitchen in a markdown file?) of all the buggy things and once we get our recipes in a good place (and good diffs vs legacy) we can fix them.